cmd/apiProviderDownload_test.go

```diff
@@ -0,0 +1,94 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type apiProviderDownloadTestUtilsBundle struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func apiProviderDownloadMockUtilsBundle() *apiProviderDownloadTestUtilsBundle {
+	utilsBundle := apiProviderDownloadTestUtilsBundle{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return &utilsBundle
+}
+
+//Successful API Provider download cases
+func TestApiProviderDownloadSuccess(t *testing.T) {
+	t.Parallel()
+	t.Run("Successful Download of API Provider", func(t *testing.T) {
+		apiServiceKey := `{
+			"oauth": {
+				"url": "https://demo",
+				"clientid": "demouser",
+				"clientsecret": "******",
+				"tokenurl": "https://demo/oauth/token"
+			}
+		}`
+
+		config := apiProviderDownloadOptions{
+			APIServiceKey:   apiServiceKey,
+			APIProviderName: "provider1",
+			DownloadPath:    "APIProvider.json",
+		}
+		// test
+		httpClient := httpMockCpis{CPIFunction: "APIProviderDownload", ResponseBody: ``, TestType: "Positive"}
+		utilsMock := apiProviderDownloadMockUtilsBundle()
+		err := runApiProviderDownload(&config, nil, &httpClient, utilsMock)
+
+		if assert.NoError(t, err) {
+
+			t.Run("Assert file download & content", func(t *testing.T) {
+				fileExist := assert.True(t, utilsMock.HasWrittenFile(config.DownloadPath))
+				if fileExist {
+					providerbyteContent, _ := utilsMock.FileRead(config.DownloadPath)
+					providerContent := string(providerbyteContent)
+					assert.Equal(t, providerContent, "{\n\t\t\t\t\"d\": {\n\t\t\t\t\t\"results\": [\n\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\"__metadata\": {\n\t\t\t\t\t\t\t\t\"id\": \"https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com:443/api/v1/MessageProcessingLogs('AGAS1GcWkfBv-ZtpS6j7TKjReO7t')\",\n\t\t\t\t\t\t\t\t\"uri\": \"https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com:443/api/v1/MessageProcessingLogs('AGAS1GcWkfBv-ZtpS6j7TKjReO7t')\",\n\t\t\t\t\t\t\t\t\"type\": \"com.sap.hci.api.MessageProcessingLog\"\n\t\t\t\t\t\t\t},\n\t\t\t\t\t\t\t\"MessageGuid\": \"AGAS1GcWkfBv-ZtpS6j7TKjReO7t\",\n\t\t\t\t\t\t\t\"CorrelationId\": \"AGAS1GevYrPodxieoYf4YSY4jd-8\",\n\t\t\t\t\t\t\t\"ApplicationMessageId\": null,\n\t\t\t\t\t\t\t\"ApplicationMessageType\": null,\n\t\t\t\t\t\t\t\"LogStart\": \"/Date(1611846759005)/\",\n\t\t\t\t\t\t\t\"LogEnd\": \"/Date(1611846759032)/\",\n\t\t\t\t\t\t\t\"Sender\": null,\n\t\t\t\t\t\t\t\"Receiver\": null,\n\t\t\t\t\t\t\t\"IntegrationFlowName\": \"flow1\",\n\t\t\t\t\t\t\t\"Status\": \"COMPLETED\",\n\t\t\t\t\t\t\t\"LogLevel\": \"INFO\",\n\t\t\t\t\t\t\t\"CustomStatus\": \"COMPLETED\",\n\t\t\t\t\t\t\t\"TransactionId\": \"aa220151116748eeae69db3e88f2bbc8\"\n\t\t\t\t\t\t}\n\t\t\t\t\t]\n\t\t\t\t}\n\t\t\t}")
```

Comment:
Please keep this simple, it is mocked anyhow. The tabs and new lines are not necessary. It could be something like:
```suggestion
					assert.Equal(t, providerContent, `{"test_content": "my content"}`)
```

Reply:
I was re-using the existing mock created for other piper steps within integration suite. If you think this is mandatory, I need to add additional mock response for this step.

Reply:
At least put this in multiple lines